src/InverseKinematics.cpp

```diff
@@ -0,0 +1,154 @@
+#include "InverseKinematics.h"
+
+
+InverseKinematics::InverseKinematics()
+:updated(false)
+,initialized(false)
+,alreadyOptimized(false)
+{
+    solverPointer = new InverseKinematicsIPOPT();
+    loader = IpoptApplicationFactory();
```

Comment:
You are probably right. 
I am not sure about the memory management needed for IPOpt. I was thinking to something like `Ipopt::SmartPtr<Ipopt::TNLP> xxx(*this);`